python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -607,3 +615,67 @@ def _showGraph(self, graph):
                 for key, refs in quantum.outputs.items():
                     dataIds = ["DataId({})".format(ref.dataId) for ref in refs]
                     print("      {}: [{}]".format(key, ", ".join(dataIds)))
+
+    def _showWorkflow(self, graph, args):
+        """Print quanta information and dependency to stdout
+
+        The input and pridicted output URIs based on the Butler repo are printed.
+        Optionally store the individual quanta to files.
+
+        Parameters
+        ----------
+        graph : `QuantumGraph`
+            Execution graph.
+        args : `argparse.Namespace`
+            Parsed command line
+        """
+        run = args.output.get("", None)
+        butler = Butler(config=args.butler_config, run=run)
+
+        iq = 0
+        hash2ParentTask = {}
+        for taskNodes in graph:
+            for quantum in taskNodes.quanta:
+                iq += 1
+                shortname = taskNodes.taskDef.taskName.split('.')[-1]
+                print("Quantum {}: {}".format(iq, shortname))
+                if args.save_qgraph:
+                    node = QuantumGraphTaskNodes(taskNodes.taskDef, [quantum],
+                                                 quantum.initInputs, quantum.outputs)
+                    qgraph = QuantumGraph()
+                    qgraph.append(node)
+                    filename = f"{args.save_qgraph}-{iq:06d}.pickle"
+                    with open(filename, "wb") as pickleFile:
+                        pickle.dump(qgraph, pickleFile)
+                print("  inputs:")
+                for key, refs in quantum.predictedInputs.items():
+                    for ref in refs:
+                        if butler.datastore.exists(ref):
+                            print("    {}".format(butler.datastore.getUri(ref)))
+                        else:
+                            fakeRef = DatasetRef(ref.datasetType, ref.dataId, run=Run(run))
+                            print("    {}".format(butler.datastore.getUri(fakeRef, predict=True)))
+                print("  outputs:")
+                for key, refs in quantum.outputs.items():
+                    for ref in refs:
+                        if butler.datastore.exists(ref):
+                            print("    {}".format(butler.datastore.getUri(ref)))
+                        else:
+                            fakeRef = DatasetRef(ref.datasetType, ref.dataId, run=Run(run))
+                            print("    {}".format(butler.datastore.getUri(fakeRef, predict=True)))
+                        # Store hash to figure out dependency
+                        dhash = hash((key, ref.dataId))
+                        hash2ParentTask[dhash] = iq
+
+        iq = 0
+        uses = []
```

Comment:
I'd probably use `set` instead of list here, searching in a set should be faster.